service/src/main/java/org/cbioportal/genome_nexus/service/remote/OncokbDataFetcher.java

```diff
@@ -0,0 +1,124 @@
+package org.cbioportal.genome_nexus.service.remote;
+
+import org.cbioportal.genome_nexus.service.exception.ResourceMappingException;
+import org.cbioportal.genome_nexus.service.transformer.ExternalResourceTransformer;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.ResourceAccessException;
+import org.springframework.web.client.RestTemplate;
+import java.util.List;
+import java.util.Map;
+import com.mongodb.BasicDBObject;
+import com.mongodb.DBObject;
+import org.oncokb.client.IndicatorQueryResp;
+
+
+@Component
+public class OncokbDataFetcher extends BaseExternalResourceFetcher<IndicatorQueryResp>
+{
+    private String oncokbToken;
+    private static final String MAIN_QUERY_PARAM = "alteration";
+    private static final String PLACEHOLDER = "PROTEINCHANGE";
+
+    private final ExternalResourceTransformer<IndicatorQueryResp> transformer;
+
+    @Autowired
+    public OncokbDataFetcher(ExternalResourceTransformer<IndicatorQueryResp> transformer,
+                                       @Value("${oncokb.url}") String oncokbUrl)
```

Comment:
it's good to put the default here as well:

```
${oncokb.url:https://www.oncokb.org/api/v1/annotate/mutations/byProteinChange?PROTEINCHANGE}
```
that way it will continue to work for people with an old properties file

Reply:
@inodb good point! will add it